### PR TITLE
fix(input, input-number): numberingSystem property works correctly

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #5252

## Summary
This PR gets `numberingSystem` working correctly for `calcite-input` and `calcite-input-number`. Previously it wouldn't let you type more than one digit in non-latin numbering systems.

- I spoke with @annierm18 and the i18n team and we do not need to support algorithmic numbering systems at this time. The numbering systems we do support are the same ones supported by [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#parameters), and are now specified in the property's type (for all components w/ the prop). 
    image
- I added a `NumberStringFormat` class with `localize` and `delocalize` methods. 
    - The class caches the `Intl.NumberFormat` formatter (assuming I know how ES modules work). 
    - Previously, 5+ formatters would be created per character typed. Now 2 formatters are created, but only when one of the `locale`, `numberingSystem`, or `groupSeparator` properties change (which is probably uncommon)
    - You will need to use the `setOptions` method before calling `localize`/`delocalize` to make sure a new formatter is created when one of the `locale`, `numberingSystem`, or `groupSeparator` properties changes. The `setOptions` call could be moved to a property watcher if that is preferred.
    - Eventually we should use this class for all number localization instead of the existing `localizeNumberString` and `delocalizeNumberString` functions (I'll create a follow up issue)

### CODE REVIEW

File: src/utils/locale.ts [hunk 1]
```diff
@@ -292,3 +329,98 @@ function getLocale(component: LocalizedComponent): string {
...
 }
+
+interface NumberStringFormatOptions {
+  numberingSystem: NumberingSystem;
+  locale: string;
+  useGrouping: boolean;
+}
+
+/**
+ * This util localizes and de-localizes numbers
```
Comment (jcfranco) [inline src/utils/locale.ts:340]: Nitpick: In the future, we should use `formats` and `parses` terminology as these are more common.

File: src/components/input/input.tsx [hunk 2]
```diff
@@ -825,40 +830,42 @@ export class Input
...
-    this.userChangedValue = origin === "user" && this.value !== newValue;
-    this.value = newValue;
+    numberStringFormatter.setOptions({
+      locale: this.effectiveLocale,
+      numberingSystem: this.numberingSystem,
+      useGrouping: this.groupSeparator
+    });
 
     if (this.type === "number") {
+      const delocalizedValue =
```
Comment (jcfranco) [inline src/components/input/input.tsx:840]: Do we need to update `input` tests to cover these changes? This applies to `input-number` as well.

**Review by jcfranco — APPROVED** [review]
Awesome stuff, @benelan! 😎

Other than some comments about an unused method and E2E test updates, this LGTM!

File: src/utils/number.ts [hunk 3]
```diff
@@ -77,6 +83,21 @@ export class BigDecimal {
     return parts;
   }
 
+  format(formatter: Intl.NumberFormat): string {
```
[the remark this answers is not in the document]
Comment (benelan) [reply src/utils/number.ts:86]: Yeah it's used in the `NumberStringFormat`'s `localize` method

[the remark this answers is not in the document]
Comment (jcfranco) [reply src/utils/number.ts:86]: Ah, gotcha. Thanks for clarifying.

**Pull request merged**